aten/src/ATen/xpu/XPUContext.cpp

```diff
@@ -0,0 +1,93 @@
+#include <ATen/xpu/XPUContext.h>
+#include <c10/util/CallOnce.h>
+#include <c10/util/Exception.h>
+
+#include <cmath>
+#include <deque>
+#include <mutex>
+#include <vector>
+
+namespace at::xpu {
+namespace {
+
+/*
+ * Currently, there is one device properties pool containing the information and
+ * capability about each compute-device.
+ *
+ * Device properties are lazily initialized when the first time properties are
+ * requested for a device.
+ */
+DeviceIndex num_gpus = -1;
+c10::once_flag init_flag;
+std::deque<c10::once_flag> device_prop_flags;
+std::vector<DeviceProp> device_properties;
+
+std::deque<c10::once_flag> device_global_idx_flags;
+std::vector<int> device_global_idxs;
```

Comment:
`device_global_idxs` is used in DLPack, so use `int` as its data type.

Reply:
Would be good to be a bit more explicit, for example use `int32_t`